packages/teleport-uidl-resolver/src/resolvers/abilities/utils.ts

```diff
@@ -43,8 +43,21 @@ export const insertLinks = (
       return node
     }
 
-    // a text node (span) on which we added a link gets transformed into an <a>
-    // the rest of the text elements get wrapped with an <a> tag
+    /* We repalce buttons with link to use <a> tag's, to make the generated
+    code to be semantically correct. */
+    if (elementType === 'button') {
+      node.content.elementType = getLinkElementType(abilities.link)
+      node.content.semanticType = ''
+      node.content.attrs = createLinkAttributes(abilities.link, options)
+      node.content.style = {
+        ...node.content.style,
+        textDecoration: { type: 'static', content: 'none' },
```

Comment:
@alexnm what are your ideas on this, should we overwrite these behaviours including, active, hover and visited too. Since they are not rendered in stage as they are basically buttons and causes inconsistencies when published !!

Reply:
not sure if this is needed (the textDecoration). when a link is set, the node inherits the style from the link style from the theme (eg: text-decoration or color if they are set). if the text in the button appears underlined, then the user should reset the underline from the inspector, hence it will appear in the UIDL.